com.unity.netcode.gameobjects/Runtime/NetworkVariable/Collections/NetworkDictionary.cs

```diff
@@ -308,17 +251,25 @@ public override bool IsDirty()
         }
 
         /// <inheritdoc />
-        public ICollection<TKey> Keys => m_Dictionary.Keys;
+        public bool TryGetValue(TKey key, out TValue value)
+        {
+            return m_Dictionary.TryGetValue(key, out value);
+        }
 
         /// <inheritdoc />
-        public ICollection<TValue> Values => m_Dictionary.Values;
+        public NativeArray<TKey> Keys(Allocator alloc)
```

Comment:
I'm not in love with this from a usability standpoint, but it seems better to me than us presuming we know what the lifetime is of 'keys' and then doing potentially unnecessary gc

Reply:
could set the default to be Temp which I assume is going to be most common use case.